README.md

```diff
@@ -67,19 +67,20 @@ Not adhering to the format will result in a `InvalidTokenError` with one of the
 - `Invalid token specified: must be a string` => the token passed was not a string, this library only works on strings. 
 - `Invalid token specified: missing part #` => this probably means you are missing a dot (`.`) in the token 
 - `Invalid token specified: invalid base64 for part #` => the part could not be base64 decoded (the message should contain the error the base64 decoder gave)
-- `Invalid token specified: invalid json for part #` => the part was correctly base64 decoded, however the decoded value was not valid json (the message should contain the error the json parser gave)
+- `Invalid token specified: invalid json for part #` => the part was correctly base64 decoded, however, the decoded value was not valid JSON (the message should contain the error the JSON parser gave)
 
 #### Use with typescript
 
-The `jwt_decode` function will return an `unknown` type by default. You can specify what the expected return type should be by passing a type argument to the `jwt_decode` function.
+The return type of the `jwt_decode` function is determined by the `header` property of the object passed as the second argument. If omitted (or set to false), it'll use `JwtPayload`, when true it will use `JwtHeader`. 
+If needed, you can specify what the expected return type should be by passing a type argument to the `jwt_decode` function.
 
-The package also exports types for a `JwtHeader` and `JwtPayload` with some default claims. You can either use them as-is, or extend them to include non standard claims or properties.
+You can extend both `JwtHeader` and `JwtPayload` to include non-standard claims or properties.
 
 ```typescript
-import jwtDecode, { JwtPayload } from "jwt-decode";
+import jwtDecode from "jwt-decode";
 
 const token: string = "eyJhsw5c";
-const decoded = jwtDecode<JwtPayload>(token); // Returns with the JwtPayload type
+const decoded = jwtDecode(token); // Returns with the JwtPayload type
```

Comment:
Could also provide an example that use the type argument or put it back altogether. It diverts attention away from the default type which while sensible, may not be academically correct.

Reply:
Looks like they had a similar discussion back then: https://github.com/auth0/jwt-decode/pull/107

Reply:
I put back the type in the readme here.